From: =?utf-8?q?Lukas_M=C3=A4rdian?= <slyon@ubuntu.com>
Date: Mon, 28 Jul 2025 15:50:19 +0200
Subject: n-dhcp4: Do not set ciaddr in DISCOVER state.

---
 shared/n-dhcp4/src/n-dhcp4-c-connection.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/shared/n-dhcp4/src/n-dhcp4-c-connection.c b/shared/n-dhcp4/src/n-dhcp4-c-connection.c
index 2f660e3..600a50b 100644
--- a/shared/n-dhcp4/src/n-dhcp4-c-connection.c
+++ b/shared/n-dhcp4/src/n-dhcp4-c-connection.c
@@ -528,6 +528,12 @@ static int n_dhcp4_c_connection_new_message(NDhcp4CConnection *connection,
         message->userdata.type = type;
         message->userdata.message_type = message_type;
 
+        /* Do not set a Client IP in DISCOVERY state */
+        if (type == N_DHCP4_C_MESSAGE_DISCOVER) {
+                message->message->header.ciaddr = INADDR_ANY;
+                connection->client_ip = INADDR_ANY;  // reset client IP
+        }
+
         /*
          * Note that some implementations expect the MESSAGE_TYPE option to be
          * the first option, and possibly even hard-code access to it. Hence,
