Index: build/src/platform/tests/test-link.c
===================================================================
--- build.orig/src/platform/tests/test-link.c
+++ build/src/platform/tests/test-link.c
@@ -475,7 +475,8 @@ test_bond (void)
 		return;
 	}
 
-	test_software (NM_LINK_TYPE_BOND, "bond");
+	g_test_skip ("Skipping test for bonding...");
+	/* test_software (NM_LINK_TYPE_BOND, "bond"); */
 }
 
 static void
